web_src/js/components/RepoActionView.vue

```diff
@@ -392,18 +392,29 @@ const sfc = {
       if (this.menuVisible) this.menuVisible = false;
     },
 
+    // show at most one of log duration and timestamp (can be both invisible)
     toggleTimeDisplay(type) {
-      if (type !== 'duration' && type !== 'stamp') return;
-      const otherType = 'durationstamp'.replace(type, '');
-      this.timeVisible[type] = !this.timeVisible[type];
-      for (const el of document.querySelectorAll(`.log-time-${type}`)) {
-        toggleElem(el, this.timeVisible[type]);
-      }
-      // show at most one of log duration and timestamp (can be both invisible)
-      if (this.timeVisible[type] && this.timeVisible[otherType]) {
-        this.timeVisible[otherType] = false;
-        for (const el of document.querySelectorAll(`.log-time-${otherType}`)) {
-          toggleElem(el, false);
+      if (type === 'stamp') {
+        this.timeVisible.showStamp = !this.timeVisible.showStamp;
+        for (const el of document.querySelectorAll('.log-time-stamp')) {
+          toggleElem(el, this.timeVisible.showStamp);
+        }
+        if (this.timeVisible.showStamp && this.timeVisible.showDuration) {
+          this.timeVisible.showDuration = false;
+          for (const el of document.querySelectorAll('.log-time-duration')) {
+            toggleElem(el, false);
+          }
+        }
+      } else {
+        this.timeVisible.showDuration = !this.timeVisible.showDuration;
+        for (const el of document.querySelectorAll('.log-time-duration')) {
+          toggleElem(el, this.timeVisible.showDuration);
+        }
+        if (this.timeVisible.showDuration && this.timeVisible.showStamp) {
+          this.timeVisible.showStamp = false;
+          for (const el of document.querySelectorAll('.log-time-stamp')) {
+            toggleElem(el, false);
+          }
```

Comment:
```

data() {
    timeVisible: {
        'log-time-stamp': false,
        'log-time-duration': false,
    }
}

toggleTimeDisplay(type) {
  const toToggleTypes = [];
  const other = type == 'duration' ? 'stamp' : 'duration';
  this.timeVisible['log-time-$type'] = !this.timeVisible['log-time-$type'];
  toToggleTypes.append(type);
  if (this.timeVisible['log-time-$other']) {
    this.timeVisible['log-time-$other'] = false
    toToggleTypes.append(other);
  }
  for (toToggle of toToggleTypes) {
    for (const el of document.querySelectorAll('.log-time-$toToggle')) {
      toggleElem(el, this.timeVisible['log-time-$toToggle']);
    } 
  }
}
```

Reply:
updated the method and used ref in https://github.com/go-gitea/gitea/pull/24876/commits/629d14d394211fa11afd97a96629d7fa929c3b80